Auto token refresh only works once per service lifetime (refreshPromise never cleared after success)

## Problem

The automatic Linear OAuth token refresh in `LinearIssueTrackerService` only works **once** per service lifetime. After the first successful refresh, all subsequent token expirations fail silently because the cached `refreshPromise` is never cleared on success.

For long-running self-hosted deployments (which is the primary use case), this means the service stops responding to Linear after the **second** token expiration (~48 hours), requiring a manual restart.

## Root Cause

In `cyrus-linear-event-transport/dist/LinearIssueTrackerService.js`, the 401 interceptor (lines 71–101):

```javascript
if (!this.refreshPromise) {
    this.refreshPromise = this.doTokenRefresh().catch((refreshError) => {
        this.refreshPromise = null;   // ← only cleared on FAILURE
        throw refreshError;
    });
}
try {
    const newToken = await this.refreshPromise;
    client.setHeader("Authorization", `Bearer ${newToken}`);
    return await client.request(document, variables, requestHeaders, true); // isRetry=true
} catch (_refreshError) {
    throw error;
}
```

After a **successful** refresh, `this.refreshPromise` remains set (a resolved Promise). When the token expires again days later:

1. `!this.refreshPromise` → `false` (old resolved Promise still cached)
2. `await this.refreshPromise` → resolves immediately with the **stale** token from the previous refresh
3. Retries with the stale token and `isRetry=true`
4. Gets 401 again, but `isRetry=true` prevents further refresh attempts → throws

The `setAccessToken()` method (line 189–192) does clear `refreshPromise`, but it's never called in the auto-refresh flow — only `client.setHeader()` is used directly.

## Reproduction

1. Self-host cyrus with `LINEAR_DIRECT_WEBHOOKS=true` and valid OAuth credentials
2. Wait for the first token to expire (~24h) — auto-refresh works correctly
3. Wait for the second token to expire (~48h) — **fails** with 401 errors, no refresh attempted

From journal logs:
```
# Apr 1 — first expiration, auto-refresh works:
[INFO ] [LinearIssueTrackerService] Refreshing token for workspace 6ee8cc0c...
[INFO ] [LinearIssueTrackerService] Token refreshed successfully for workspace 6ee8cc0c...

# Apr 5 — second expiration, no refresh attempt, just errors:
[ERROR] [EdgeWorker] Failed to fetch issue description for routing: Authentication required
[ERROR] [RepositoryRouter] Failed to fetch labels for routing: Authentication required
```

## Suggested Fix

Clear `refreshPromise` after the token is successfully obtained, before retrying:

```javascript
try {
    const newToken = await this.refreshPromise;
    this.refreshPromise = null;  // ← add this line
    client.setHeader("Authorization", `Bearer ${newToken}`);
    return await client.request(document, variables, requestHeaders, true);
}
```

This is safe because `doTokenRefresh()` already uses the workspace-level `pendingRefreshes` static Map for coalescing concurrent refresh requests.

## Environment

- cyrus-ai: 0.2.38 (also verified unchanged in 0.2.40)
- cyrus-linear-event-transport: latest bundled version
- Platform: self-hosted Linux, `LINEAR_DIRECT_WEBHOOKS=true`
- Node: v22.22.0

## Workaround

Manually patch the installed file or periodically restart the cyrus service.

--- a/packages/linear-event-transport/src/LinearIssueTrackerService.ts
+++ b/packages/linear-event-transport/src/LinearIssueTrackerService.ts
@@ -143,9 +143,7 @@ export class LinearIssueTrackerService implements IIssueTrackerService {
 					// or if it's not a token expiration error
 					if (isRetry || !this.isTokenExpiredError(error)) throw error;
 
-					// Coalesce ALL concurrent refresh attempts - everyone shares the same promise.
-					// The promise persists after resolution so late-arriving 401s still get
-					// the same token without triggering a new refresh.
+					// Coalesce concurrent refresh attempts - everyone shares the same promise.
 					if (!this.refreshPromise) {
 						this.refreshPromise = this.doTokenRefresh().catch(
 							(refreshError) => {
@@ -159,6 +157,9 @@ export class LinearIssueTrackerService implements IIssueTrackerService {
 
 					try {
 						const newToken = await this.refreshPromise;
+						// Clear cached promise so future token expirations trigger a fresh refresh.
+						// Workspace-level coalescing via pendingRefreshes still deduplicates concurrent calls.
+						this.refreshPromise = null;
 						client.setHeader("Authorization", `Bearer ${newToken}`);
 
 						// Retry the request with the new token (marked as retry to prevent loops)

